packages/mui-base/src/utils/useAnimationUnmount.ts

```diff
@@ -0,0 +1,55 @@
+'use client';
+import * as React from 'react';
+import { useEventCallback } from './useEventCallback';
+import { ownerWindow } from './owner';
+
+/**
+ * Unmounts the supplied element only if it has no animations or transitions.
+ * @ignore - internal hook.
+ */
+export function useAnimationUnmount(
```

Comment:
I find it quite hard to understand the purpose of this function when reading the code. It doesn't really unmount anything (as the name would suggest), but it's a means of detecting if the element has any animation defined in its styles. Renaming it could make it easier to grasp.

Reply:
Yeah, I didn't really know what to call it without it being really long. It returns an `unmount` function, but it's simply guarded to only run if the element has no animations/transitions.

- `useUnmountIfNoAnimations`